src/QsCompiler/Tests.Compiler/TestCases/LocalVerification.qs

```diff
@@ -51,6 +51,144 @@ namespace Microsoft.Quantum.Testing.LocalVerification {
         using ((q1, q2, q3) = (Qubit(), (Qubit(), Qubit()))) {}
     }
 
+    operation VariableDeclaration11<'T>(cnt: Int, arg : 'T) : Unit {
+        let recur = VariableDeclaration11; // not allowed
+        recur(cnt - 1, arg);
+    }
+
+    operation VariableDeclaration12<'T>(cnt: Int, arg : 'T) : Unit {
+        let tuple = (1, (VariableDeclaration12, "")); // not allowed
+        let (_, recur) = tuple;
+        recur(cnt - 1, arg);
+    }
+
+    operation VariableDeclaration13<'T>(cnt: Int, arg : 'T) : Unit {
+        mutable arr = new ((Int, 'T) => Unit)[0];
+        set arr += [VariableDeclaration12]; 
```

Comment:
```suggestion
        set arr += [VariableDeclaration13]; 
```

Reply:
I think this was meant to be a recursive reference, right?

Reply:
No, that is covered in a test further below.